src/LineChart/components/LineChart.ts

```diff
@@ -0,0 +1,129 @@
+import { Component, createElement } from "react";
+
+import * as classNames from "classnames";
+import { Config, LineLayout, PlotlyStatic, ScatterData } from "plotly.js";
+
+import "../ui/LineChart.css";
+
+declare function require(name: string): string;
+
+interface LineChartProps {
+    data?: ScatterData[];
+    config?: Partial<Config>;
+    layout?: Partial<LineLayout>;
+    width: number;
+    widthUnit: string;
+    height: number;
+    heightUnit: string;
+    className?: string;
+    style?: object;
+}
+
+class LineChart extends Component<LineChartProps, {}> {
+    private lineChart: HTMLDivElement;
+    private Plotly: PlotlyStatic;
+    private data: ScatterData[] = [
+        {
+            connectgaps: true,
+            mode: "lines+markers",
+            name: "Default data points",
+            type: "scatter",
+            x: [ 14, 20, 30, 50 ],
+            y: [ 14, 30, 20, 40 ]
+        }
+    ];
+
+    constructor(props: LineChartProps) {
+        super(props);
+
+        this.Plotly = require("plotly.js/dist/plotly") as any;
+
+        this.getPlotlyNodeRef = this.getPlotlyNodeRef.bind(this);
+        this.onResize = this.onResize.bind(this);
+    }
+
+    render() {
+        return createElement("div", {
+            className: classNames("widget-line-chart", this.props.className),
+            ref: this.getPlotlyNodeRef,
+            style: {
+                ...this.props.style,
+                ...this.getStyle()
+            }
+        });
+    }
+
+    componentDidMount() {
+        this.renderChart(this.props);
+        this.setUpEvents();
+        this.adjustStyle();
+    }
+
+    componentWillReceiveProps(newProps: LineChartProps) {
+        this.renderChart(newProps);
+    }
+
+    componentWillUnmount() {
+        if (this.lineChart) {
+            this.Plotly.purge(this.lineChart);
+        }
+        window.removeEventListener("resize", this.onResize);
+    }
+
+    private getPlotlyNodeRef(node: HTMLDivElement) {
+        this.lineChart = node;
+    }
+
+    private adjustStyle() {
+        if (this.lineChart) {
+            const wrapperElement = this.lineChart.parentElement;
+            if (this.props.heightUnit === "percentageOfParent" && wrapperElement) {
+                wrapperElement.style.height = "100%";
```

Comment:
This makes sure the parent is always at 100% so that the child height can be set to percentage of the height